shell.go

```diff
@@ -62,14 +64,28 @@ func NewLocalShell() *Shell {
 }
 
 func NewShell(url string) *Shell {
-	c := &gohttp.Client{
-		Transport: &gohttp.Transport{
-			Proxy:             gohttp.ProxyFromEnvironment,
-			DisableKeepAlives: true,
-		},
+	var client *http.Client
+
+	tpt := &http.Transport{
```

Comment:
Let's move all of this logic down into `NewShellWithClient`, overriding the dial function (when unset).